### PR TITLE
fix: improve YouTube watch url parser

### PR DESCRIPTION
**Pull request by mcanouil**
## Description

This PR:
- improve the matching pattern for URL like `https://www.youtube.com/watch?v=wo9vZccmqwc` (see #3781).
- remove a duplicated test.
  https://github.com/quarto-dev/quarto-cli/blob/80548e60915e7c77ee39e049aa1a0eee1704730d/src/resources/extensions/quarto/video/_tests/test-suite.lua#L60-L75
- add a test for `https://www.youtube.com/watch?v=wo9vZccmqwc` with no additional arguments in the URL.

## Checklist

I have (if applicable):

- [x] filed a [contributor agreement](../CONTRIBUTING.md).
- [x] referenced the GitHub issue this PR closes
- [ ] updated the appropriate changelog

### CODE REVIEW

**Comment by cscheid** [comment]
Do note that the test wasn't duplicated:

```
  match = match or string.match(src, '%?v=(.-)&')
  match = match or string.match(src, '%?v=(.-)$')
```

`&` vs `$`, specifically.

**Comment by mcanouil** [comment]
> Do note that the test wasn't duplicated:
> 
> ```
>   match = match or string.match(src, '%?v=(.-)&')
>   match = match or string.match(src, '%?v=(.-)$')
> ```
> 
> `&` vs `$`, specifically.

Oh, I missed that.
The test I added is more reliable, but now I wonder did it really fix the issue.

**Comment by mcanouil** [comment]
After few testing, the issue I tried to fix in #3824 is actually the same as with brightcove URL as @cscheid described in #3744
I reverted the commit on the changelog.

The other changes are still "good" for performance/testing but let me know, if I close this PR for you to make one whole PR to actually fix the key=value parsing issue.

File: src/resources/extensions/quarto/video/_tests/test-suite.lua [hunk 1]
```diff
@@ -65,9 +65,9 @@ function TestYouTubeBuilder:testShareURL()
   checkYouTubeBuilder(params, expected)
 end
 
-function TestYouTubeBuilder:testShareURL()
+function TestYouTubeBuilder:testWebRawURL()
```
Comment (allenmanning) [inline src/resources/extensions/quarto/video/_tests/test-suite.lua:68]: Thanks for the fix here.  Yep this is a duplicate.

File: src/resources/extensions/quarto/video/video.lua [hunk 2]
```diff
@@ -69,8 +69,7 @@ local youTubeBuilder = function(params)
   local src = params.src
   match = checkEndMatch(src, 'https://www.youtube.com/embed/')
   match = match or checkEndMatch(src, 'https://youtu.be/')
-  match = match or string.match(src, '%?v=(.-)&')
-  match = match or string.match(src, '%?v=(.-)$')
+  match = match or string.match(src, '%?v=([%w-_]+)')
```
Comment (allenmanning) [inline src/resources/extensions/quarto/video/video.lua:72]: 👍

**Review by ghost — APPROVED** [review]
💯 Thank you @mcanouil

**Comment by cscheid** [comment]
I'm going to close this since we've improved the handling of key=value vs raw_args parsing. In any case, thanks for the PR, @mcanouil .

**Pull request closed without merging**